packages/components/inputs/selectable-search-input/src/selectable-search-input.tsx

```diff
@@ -89,6 +90,10 @@ export type TSelectableSearchInputProps = {
    * Value of the input. Consists of text input and selected option.
    */
   value: TValue;
+  /**
+   * Default value of the input. Consists of text input and selected option.
+   */
+  defaultValue: TValue;
```

Comment:
Is this supposed to be a required field?

If yes, it should probably be considered a breaking change.
If not, it should be handled as an optional field.

Reply:
Oh true, seems to me like it should be optional, assuming we do not expect users to always provide a default value.
The initial implementation already sets it to an empty string in the absence of a default value.

Reply:
The problem I see is that we currently only have the `value` property but we're actually using it as default value.
If we now change the behaviour of the already exposed `value` property to drive the internal state and add the new `defaultValue`, all current consumers will be affected so hence this would be a breaking change as Nicola mentioned.

We need to double think if we can somehow come up with a solution that can avoid the breaking change.

Reply:
Nicola and I had a conversation about this and we agreed releasing a major `ui-kit` version just for this change currently is not convinient.
Although the long term solution is having both a `defaultValue` and a `value` properties to handle both controlled and uncontrolled states, we think it could benefit us to hold that change until we have a better reason to release a breaking change version.

What we can do in the meantime is to introduce a new temporary property (example: `experimentalValue`) that can be used for the controlled version of the component. Then, we will have the two properties that we need, but with the "wrong" names and at least we will have a escape hatch for those (very few) use cases we know so far that require the controlled version of the component.

Once we get to the point when it will be convenient to release an `ui-kit` major version, we will rename those properties and probably publish a _codemod_ which consumers can run in their codebase, to migrate from one names to the others automatically.